Sources/ArcGISToolkit/Components/Offline/OfflineMapAreasView.swift

```diff
@@ -84,31 +82,64 @@ public struct OfflineMapAreasView: View {
             } else {
                 emptyPreplannedMapAreasView
             }
-        case .failure(let error):
-            VStack(alignment: .center) {
-                Image(systemName: "exclamationmark.circle")
-                    .imageScale(.large)
-                    .foregroundStyle(.red)
-                Text(error.localizedDescription)
+        case .failure(let error as URLError):
+            if error.code == .notConnectedToInternet {
+                offlinePreplannedMapAreasView
+            } else {
+                view(for: error)
             }
-            .frame(maxWidth: .infinity)
+        case .failure(let error):
+            view(for: error)
```

Comment:
These are really both the same case (failure), so they should be combined and the error type check should be moved inside the switch case:

```swift
case .failure(let error):
    if let urlError = error as? URLError,
       urlError.code == .notConnectedToInternet {
        offlinePreplannedMapAreasView
    } else {
        view(for: error)
    }
```